### ISSUE TITLE
NavigationMapRoute Crash after closing NavigationView activity

### ISSUE THREAD

**Issue by nchorniy** [mapbox/mapbox-navigation-android#268]
I was running a demo app and after exiting navigation launched by `NavigationLauncher.startNavigation`
sporadically app was crashing. Logs attached:

```
09-25 19:47:21.921 18697-18697/com.mapbox.services.android.navigation.testapp E/NativeMapView: You're calling `getSource` after the `MapView` was destroyed, were you invoking it after `onDestroy()`?
09-25 19:47:21.941 18697-18697/com.mapbox.services.android.navigation.testapp E/NativeMapView: You're calling `addSource` after the `MapView` was destroyed, were you invoking it after `onDestroy()`?
09-25 19:47:21.941 18697-18697/com.mapbox.services.android.navigation.testapp E/NativeMapView: You're calling `getLayers` after the `MapView` was destroyed, were you invoking it after `onDestroy()`?
09-25 19:47:21.941 18697-18697/com.mapbox.services.android.navigation.testapp D/AndroidRuntime: Shutting down VM
09-25 19:47:21.941 18697-18697/com.mapbox.services.android.navigation.testapp E/AndroidRuntime: FATAL EXCEPTION: main
                                                                                                Process: com.mapbox.services.android.navigation.testapp, PID: 18697
                                                                                                java.lang.NullPointerException: Attempt to invoke interface method 'java.util.Iterator java.util.List.iterator()' on a null object reference
                                                                                                    at com.mapbox.services.android.navigation.ui.v5.route.NavigationMapRoute.setLayerVisibility(NavigationMapRoute.java:271)
                                                                                                    at com.mapbox.services.android.navigation.ui.v5.route.NavigationMapRoute.addRoute(NavigationMapRoute.java:203)
                                                                                                    at com.mapbox.services.android.navigation.ui.v5.NavigationView.startNavigation(NavigationView.java:603)
                                                                                                    at com.mapbox.services.android.navigation.ui.v5.NavigationView.onResponse(NavigationView.java:299)
                                                                                                    at retrofit2.ExecutorCallAdapterFactory$ExecutorCallbackCall$1$1.run(ExecutorCallAdapterFactory.java:68)
                                                                                                    at android.os.Handler.handleCallback(Handler.java:739)
                                                                                                    at android.os.Handler.dispatchMessage(Handler.java:95)
                                                                                                    at android.os.Looper.loop(Looper.java:158)
                                                                                                    at android.app.ActivityThread.main(ActivityThread.java:7230)
                                                                                                    at java.lang.reflect.Method.invoke(Native Method)
                                                                                                    at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:1230)
                                                                                                    at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1120)
```

**Comment by nchorniy** [comment]
It was happening on release v0.6.0

**Comment by cammace** [comment]
Can you provide your activities lifecycle snippet of code? Sounds like you are calling `mapView.onDestroy` before destroying a map plugin.

**Comment by dmateus** [comment]
I'm having the same issue. I have both a mapView shown inside a fragment and a navigation screen that the user might start. 

How do I correctly destroy the navigation screen started by:
`NavigationLauncher.startNavigation(...)`

**Comment by danesfeder** [comment]
@nchorniy @dmateus Is this an intermittent issue you're seeing?  I haven't been able to reproduce this myself.  This happens when the UI closes or the app that is running it is destroyed?   

We may need to introduce a lifecycle method in `NavigationMapRoute` to ensure it's properly destroyed.

**Comment by danesfeder** [comment]
Fixed in #484
